connect/transforms/src/main/java/org/apache/kafka/connect/transforms/field/FieldPath.java

```diff
@@ -0,0 +1,377 @@
+        } else {
+            if (PATHS_CACHE.containsKey(field)) {
+                return PATHS_CACHE.get(field);
+            } else {
+                final FieldPath fieldPath = new FieldPath(field, version);
+                PATHS_CACHE.put(field, fieldPath);
+                return fieldPath;
+            }
+        }
+    }
+
+    FieldPath(String path, FieldSyntaxVersion version) {
+        if (path == null || path.isEmpty()) { // empty path
+            this.path = new String[] {};
+        } else {
+            switch (version) {
+                case V1: // backward compatibility
+                    this.path = new String[] {path};
+                    break;
+                case V2:
+                    // if no dots or wrapping backticks are used, then return path with single step
+                    if (!path.contains(DOT)
+                        && !(path.startsWith(BACKTICK) && path.endsWith(
+                        BACKTICK))) {
+                        this.path = new String[] {path};
+                    } else {
+                        // prepare for tracking path steps
+                        final List<String> steps = new ArrayList<>();
+                        final StringBuilder s = new StringBuilder(
+                            path); // avoid creating new string on changes
+
+                        while (s.length() > 0) { // until path is traverse
+                            // process backtick pair if any
+                            if (s.charAt(0) == BACKTICK_CHAR) {
+                                s.deleteCharAt(0);
+
+                                // find backtick pair
+                                int idx = 0;
+                                while (idx >= 0) {
+                                    idx = s.indexOf(BACKTICK, idx);
+                                    if (idx == -1) {
+                                        throw new IllegalArgumentException(
+                                            "Incomplete backtick pair at [...]`" + s);
+                                    }
+                                    if (idx != s.length() - 1) { // non-global backtick
+                                        if (s.charAt(idx + 1) != DOT_CHAR
+                                            || s.charAt(idx - 1)
+                                            == BACKSLASH_CHAR) { // not wrapped or escaped
+                                            idx++; // move index forward and keep searching
+                                        } else { // it's end pair
+                                            steps.add(
+                                                checkIncompleteBacktickPair(s.substring(0, idx)));
+                                            s.delete(0, idx + 2); // rm backtick and dot
+                                            break;
+                                        }
+                                    } else { // global backtick
```

Comment:
Is a "global" backtick referring to a pair of backticks that wrap the whole field path? I didn't get why that requires special handling?

Reply:
Not really. I was trying to optimize for that scenario; though there is not really much to optimize for haha. I removed the concept to avoid confusion.